pkgs/circuits/halo2-secp256k1-wasm/src/eff_ecdsa.rs

```diff
@@ -0,0 +1,402 @@
+#![allow(non_snake_case)]
+use anyhow::{Context, Result};
+use halo2_base::{
+    gates::{circuit::builder::BaseCircuitBuilder, RangeChip},
+    halo2_proofs::halo2curves::{
+        bn256::Fr as Bn254Fr,
+        secp256k1::{Fp as Secp256k1Fp, Fq as Secp256k1Fq, Secp256k1Affine},
+    },
+    utils::{BigPrimeField, CurveAffineExt},
+    Context as Halo2Context,
+};
+use halo2_ecc::{
+    bigint::ProperCrtUint,
+    ecc::{EcPoint, EccChip},
+    fields::{fp::FpChip, FieldChip},
+    secp256k1::{FpChip as Secp256k1FpChip, FqChip as Secp256k1FqChip},
+};
+use halo2_wasm::Halo2Wasm;
+use std::{cell::RefCell, rc::Rc};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[derive(Debug)]
+pub struct EffECDSAInputs {
+    s: Secp256k1Fq,
+    pk: Secp256k1Affine,
+    T: Secp256k1Affine,
+    U: Secp256k1Affine,
+}
+
+#[wasm_bindgen]
+pub struct Secp256k1VerifyCircuit {
+    eff_ecdsa_inputs: EffECDSAInputs,
+    range_chip: RangeChip<Bn254Fr>,
+    builder: Rc<RefCell<BaseCircuitBuilder<Bn254Fr>>>,
+}
+
+impl Secp256k1VerifyCircuit {
+    pub fn new(halo2_wasm: &Halo2Wasm, ecdsa_inputs: EffECDSAInputs) -> Result<Self> {
+        let ecdsa_inputs = EffECDSAInputs {
+            s: ecdsa_inputs.s,
+            pk: ecdsa_inputs.pk,
+            T: ecdsa_inputs.T,
+            U: ecdsa_inputs.U,
+        };
+
+        let circuit_params = halo2_wasm
+            .circuit_params
+            .clone()
+            .context("Error: Circuit params are not set")?;
+
+        let lookup_bits = circuit_params
+            .lookup_bits
+            .context("Error: Lookup bits are not set in circuit params")?;
+
+        let range = RangeChip::new(
+            lookup_bits,
+            halo2_wasm.circuit.borrow().lookup_manager().clone(),
+        );
+
+        Ok(Secp256k1VerifyCircuit {
+            eff_ecdsa_inputs: ecdsa_inputs,
+            range_chip: range,
+            builder: Rc::clone(&halo2_wasm.circuit),
+        })
+    }
+
+    fn secp256k1_fp_chip(&self) -> Secp256k1FpChip<Bn254Fr> {
+        let limb_bits = 88;
+        let num_limbs = 3;
+        Secp256k1FpChip::<Bn254Fr>::new(&self.range_chip, limb_bits, num_limbs)
+    }
+    fn secp256k1_fq_chip(&self) -> Secp256k1FqChip<Bn254Fr> {
+        let limb_bits = 88;
+        let num_limbs = 3;
+        Secp256k1FqChip::<Bn254Fr>::new(&self.range_chip, limb_bits, num_limbs)
+    }
+
+    fn ecc_chip<'a, Fp: BigPrimeField>(
```

Comment:
to please the linter (I think we discussed keeping the generics to support other curves later, even if we don't need the generics now)
```suggestion
    #[allow(clippy::extra_unused_type_parameters)]
    fn ecc_chip<'a, Fp: BigPrimeField>(
```

Reply:
The generics are done in another PR.